### PR TITLE
Feature/cplusplus/carlosc

### PR DESCRIPTION
**Pull request by carlos8882**
C++ code

### CODE REVIEW

**Review by carlosmc23 — COMMENTED** [review]
I think is not necesary upload another main class

File: src/main/java/org/jalasoft/moi/model/core/IHandler.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+package org.jalasoft.moi.model.core;
+/**
+ * Handler is in charge to work with the executer and the command builder
+ *
+ * @version 1.0
+ * @author Carlos Meneses
+ */
+public interface IHandler {
+
+
```
Comment (nanrehd) [inline src/main/java/org/jalasoft/moi/model/core/IHandler.java]: Please remove oneblank line.

File: src/main/java/org/jalasoft/moi/model/core/IHandler.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ *   Copyright (c) 2020 Jalasoft.
+ *
+ *   This software is the confidential and proprietary information of Jalasoft.
+ *   ("Confidential Information"). You shall not disclose such Confidential
+ *   Information and shall use it only in accordance with the terms of the
+ *   license agreement you entered into with Jalasoft.
+ */
+
+package org.jalasoft.moi.model.core;
+/**
```
Comment (nanrehd) [inline src/main/java/org/jalasoft/moi/model/core/IHandler.java]: Please add a blank line.

File: src/main/java/org/jalasoft/moi/controller/CppCompile.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+ *
+ * This software is the confidential and proprietary information of Jalasoft.
+ * ("Confidential Information"). You shall not disclose such Confidential
+ * Information and shall use it only in accordance with the terms of the
+ * license agreement you entered into with Jalasoft.
+ */
+
+package org.jalasoft.moi.controller;
+
+import java.io.*;
```
Comment (nanrehd) [inline src/main/java/org/jalasoft/moi/controller/CppCompile.java]: Please don't use '*' for imports.

**Review by nanrehd — COMMENTED** [review]
Please use all our standards known. Great job on the command builder.

**Review by nanrehd — DISMISSED** [review]

**Review by nanrehd — APPROVED** [review]

**Review by carlosmc23 — COMMENTED** [review]
i think you sould not use a push force in your pull request

File: src/main/java/org/jalasoft/moi/model/cplusplus/CppCommandBuilder.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+ * ("Confidential Information"). You shall not disclose such Confidential
+ * Information and shall use it only in accordance with the terms of the
+ * license agreement you entered into with Jalasoft.
+ */
+package org.jalasoft.moi.model.cplusplus;
+
+import org.jalasoft.moi.model.core.ICommandBuilder;
+
+import java.nio.file.Path;
+
```
Comment (carlosmc23) [inline src/main/java/org/jalasoft/moi/model/cplusplus/CppCommandBuilder.java:14]: you need comment your class, please check java standars

File: src/test/java/org/jalasoft/moi/model/cpp/CppCommandBuilderTest.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+import org.jalasoft.moi.model.core.Language;
+import org.jalasoft.moi.model.core.Params;
+import org.jalasoft.moi.model.cplusplus.CppCommandBuilder;
+import org.jalasoft.moi.model.csharp.CsharpCommandBuilder;
+import org.junit.jupiter.api.Test;
+
+import java.nio.file.Paths;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
```
Comment (carlosmc23) [inline src/test/java/org/jalasoft/moi/model/cpp/CppCommandBuilderTest.java:22]: you need comment this

**Pull request merged**